src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -639,8 +668,28 @@ _mongoc_client_session_handle_reply (mongoc_client_session_t *session,
          mongoc_client_session_advance_cluster_time (session, &cluster_time);
       } else if (!strcmp (bson_iter_key (&iter), "operationTime") &&
                  BSON_ITER_HOLDS_TIMESTAMP (&iter) && is_acknowledged) {
-         bson_iter_timestamp (&iter, &t, &i);
-         mongoc_client_session_advance_operation_time (session, t, i);
+         bson_iter_timestamp (&iter, &operation_t, &operation_i);
+         mongoc_client_session_advance_operation_time (
+            session, operation_t, operation_i);
+      } else if (!strcmp (bson_iter_key (&iter), "cursor") &&
```

Comment:
Based on the spec:

> Saving the ``atClusterTime`` returned by 5.0+ servers for the first find/aggregate/distinct

I think we should limit the parsing to replies from `find` / `aggregate` / `distinct`. I think that `_mongoc_client_session_handle_reply` could accept a `cmd_name` argument, and the caller, `mongoc_cluster_run_opmsg`, can pass `cmd->command_name`.

Reply:
Good point, otherwise we'll be doing looking at `cursor` replies for everything. Made this conditional on `is_find_aggregate_distinct` and added new `cmd_name` param to `_mongoc_client_session_handle_reply`.